src/main/java/org/springframework/data/couchbase/repository/query/StringBasedN1qlQueryParser.java

```diff
@@ -196,25 +208,68 @@ private boolean checkNotQuoted(String item, int start, int end, List<int[]> quot
 	private JsonValue getPositionalPlaceholderValues(ParameterAccessor accessor) {
 		JsonArray posValues = JsonArray.create();
 		for (Parameter parameter : this.queryMethod.getParameters().getBindableParameters()) {
-			posValues.add(this.couchbaseConverter.convertForWriteIfNeeded(accessor.getBindableValue(parameter.getIndex())));
+			Object value = couchbaseConverter.convertForWriteIfNeeded(accessor.getBindableValue(parameter.getIndex()));
+				try {
+					posValues.add(value);
+				}catch(InvalidArgumentException iae){
+					Object o = accessor.getBindableValue(parameter.getIndex());
+					if(o instanceof Object[]) { // Maybe just need to treat it as an array ?
+						Object[] array=(Object[])o;
+						JsonArray ja=JsonValue.ja();
+						for(Object e:array) {
+							ja.add(String.valueOf(couchbaseConverter.convertForWriteIfNeeded(e)));
+						}
+						posValues.add(ja);
+					} else {
+						throw iae;
+					}
+				}
 		}
 		return posValues;
 	}
 
 	private JsonObject getNamedPlaceholderValues(ParameterAccessor accessor) {
 		JsonObject namedValues = JsonObject.create();
-
+		HashSet<String> pNames=new HashSet<>(parameterNames);
 		for (Parameter parameter : this.queryMethod.getParameters().getBindableParameters()) {
 			String placeholder = parameter.getPlaceholder();
-			Object value = this.couchbaseConverter.convertForWriteIfNeeded(accessor.getBindableValue(parameter.getIndex()));
+			Object value = couchbaseConverter.convertForWriteIfNeeded(accessor.getBindableValue(parameter.getIndex()));
 
 			if (placeholder != null && placeholder.charAt(0) == ':') {
 				placeholder = placeholder.replaceFirst(":", "");
-				namedValues.put(placeholder, value);
+				try {
```

Comment:
Another instance of deep nesting without proper formatting.

Please extract these at least into some sort of utility methods.